net-core/api/src/main/java/org/pragmatica/net/http/HttpFunction.java

```diff
@@ -0,0 +1,515 @@
+///           .as(APPLICATION_JSON);
+///
+/// // Define endpoint with request body
+/// Fn2<Promise<User>, UserId, UpdateUserRequest> updateUser = 
+///     client.function("https://api.example.com")
+///           .path("api/v1/users").pathVar(UserId.class)
+///           .send(APPLICATION_JSON, UpdateUserRequest.class)
+///           .put(User.class)
+///           .as(APPLICATION_JSON);
+///
+/// // Use multiple times with different parameters
+/// var user1 = getUser.apply(new UserId("123")).await();
+/// var updatedUser = updateUser.apply(new UserId("123"), updateRequest).await();
+/// }
+public interface HttpFunction {
+    
+    /// Start building a function-style endpoint with path segments
+    HttpFunctionBuilder0 path(String pathSegments);
+    
+    /// Function builder with no path variables - path building and content type bridge
+    interface HttpFunctionBuilder0 {
+        
+        // === Path Building ===
+        
+        /// Add more path segments
+        HttpFunctionBuilder0 path(String pathSegments);
+        
+        /// Add a typed path variable
+        <T1> HttpFunctionBuilder1<T1> pathVar(Class<T1> type);
+        
+        /// Add a generic typed path variable
+        <T1> HttpFunctionBuilder1<T1> pathVar(TypeToken<T1> type);
+        
+        // === Client Configuration ===
+        
+        /// Set connection timeout
+        HttpFunctionBuilder0 connectTimeout(org.pragmatica.lang.io.TimeSpan timeout);
+        
+        /// Set request timeout
+        HttpFunctionBuilder0 requestTimeout(org.pragmatica.lang.io.TimeSpan timeout);
+        
+        /// Set read timeout
+        HttpFunctionBuilder0 readTimeout(org.pragmatica.lang.io.TimeSpan timeout);
+        
+        /// Set user agent
+        HttpFunctionBuilder0 userAgent(String userAgent);
+        
+        /// Add default header
+        HttpFunctionBuilder0 header(String name, String value);
+        
+        /// Set default headers
+        HttpFunctionBuilder0 headers(HttpHeaders headers);
+        
+        /// Set max connections
+        HttpFunctionBuilder0 maxConnections(int maxConnections);
+        
+        /// Set max connections per host
+        HttpFunctionBuilder0 maxConnectionsPerHost(int maxConnectionsPerHost);
+        
+        /// Set follow redirects
+        HttpFunctionBuilder0 followRedirects(boolean followRedirects);
+        
+        // === Content Type Bridge Methods ===
+        
+        /// Send request with no body - specify request content type for Accept header
+        HttpMethodBuilder0 send(ContentType requestContentType);
+        
+        /// Send JSON request with no body - convenience method
+        default HttpMethodBuilder0 sendJson() {
+            return send(CommonContentTypes.APPLICATION_JSON);
+        }
+        
+        /// Send request with body - specify request content type and body type
+        <T> HttpMethodBuilderWithBody0<T> send(ContentType requestContentType, Class<T> bodyType);
```

Comment:
Actually you can forward all calls which accept Class<T> to the corresponding methods which use TypeToken using TypeToken.of() to wrap Class into TypeToken. This should significantly reduce implementation.